lib/jsonapi/authorization/default_pundit_authorizer.rb

```diff
@@ -77,8 +77,10 @@ def show_related_resource(source_record:, related_record:)
       # ==== Parameters
       #
       # * +source_record+ - The record whose relationship is queried
-      def show_related_resources(source_record:)
+      # * +related_record_class+ - The associated record class to show
+      def show_related_resources(source_record:, related_record_class:)
         ::Pundit.authorize(user, source_record, 'show?')
+        ::Pundit.authorize(user, related_record_class, 'index?')
```

Comment:
Is there a scenario where `related_record_class` could be `nil`? The code currently using `safe_constantize` would mean that it could return `nil`, which would cause a `nil` error to be raised on this line

Reply:
I now use the `@resource_klass` attribute. I am not sure if it could be nil but I would execpt that it cannot be nil since you ask for the related models so it would be strange if the related class is nil since you request for it

Reply:
Sounds reasonable to me! I like your solution :+1: